docs/development/dsl.md

```diff
@@ -0,0 +1,49 @@
+# Trento Checks DSL Specification
+
+## Foreword
+The need this DSL aims to fulfill is to provide users a simple way to declare what we (the Trento Core Team) often refer to as "checks". Checks are, in Trento's domain, the crystallization of SUSE's best practices when it comes to SAP clusters configuration in a form that both a man and a machine can read.
+
+We get several benefits from this approach:
+
+- Humans can formalize best practices with no space for ambiguity;
+- Machines can assert systems' state, automatically, with no space for ambiguity;
+- The development of new best practices gets streamlined through a common definition that allows to firestart shared efforts.
+
+## Facts
+Fact declaration and gathering revolves all around the concept of _gatherer_. Gatherers are functions directly built inside the interpretation level of the language, and reflect the various types of information that the platform can discover on the hosts, and on a given cluster. Examples include (but are not limited to) installed packages, open ports, and configuration files' content. Gathered facts can be matched against values inside expectations, using the name that we give them inside the fact declaration.
+
+A fact gathering declaration contains:
+
+- The fact name;
+- The function used to gather the fact;
```

Comment:
I'd write this as:

`- The pre-defined gatherer function used for this fact`



(Not that it's wrong as it is, just maybe to use the `gatherer` term again instead of function)


Nitpicking, I know :smile: